app/backend/src/app.py

```diff
@@ -73,6 +74,8 @@ def log_unhandled_exception(exc_type, exc_value, exc_traceback):
 
 apply_migrations()
 
+update_data()
```

Comment:
The migrations create the table, etc. `update_constant_data` updates the actual table data to match what's in the git source of truth.

Reply:
Can't we migrate constant tables? I don't see the semantic difference between a table lookup or a function that does the same thing, and functions are added by migrations.

Reply:
I agree that the conceptual difference is small, and the update function is a bit ugly.

On the other hand, this is much simpler: say we launch the beta and we notice we missed a language that some people would like to add. This allows Itsi to go and update the JSON and add a few lines, instead of someone with more backend experience having to set up the backend and create migrations. If this happens a few times, it'll get a bit annoying. This will also cause fewer migrations which slow down the tests, and most tests don't use the languages.

Reply:
Even simpler is that they just enter the three-letter code in the frontend language and we have no checks in the backend.

Reply:
Or move the check that the backend does to Python.

Reply:
> Even simpler is that they just enter the three-letter language code in the frontend and we have no checks in the backend.

The backend needs to be aware of the list because we need different clients to agree on what the allowed languages are. Sure, we could probably synchronize the official clients by just pulling the JSON from git or something, but what about 3rd party clients? I don't like the idea of storing gibberish data in the database also. What if someone hits the API directly and just adds rubbish?

> Or move the check that the backend does to Python.

Parse the JSON and store it in a hash map throughout the life of the app? Why bother when we have an excellent database that can enforce data integrity and has built-in indexing and fast lookups, etc. I think postgres is the right tool for the job!

[Although it's not a good argument, I'm also weary of the fact that we will probably soon need to add more complex constant tables. On the table (no pun intended) are regions/countries (may include a geo component), or timezone polygons (has geo component which definitely needs to be done in the db).]

Reply:
> What if someone hits the API directly and just adds rubbish?

Then their profile will look like rubbish, nothing else.

> Why bother when we have an excellent database that can enforce data integrity

...without proper migrations so that the backend fails to start when we remove a previously allowed language.

Reply:
I asked on the postgres slack. We could also just straight up read the source of truth CSV through a file foreign data wrapper in postgres: https://www.postgresql.org/docs/current/file-fdw.html

Reply:
There seem to be 7893 languages in [ISO639-3](https://iso639-3.sil.org/sites/iso639-3/files/downloads/iso-639-3.tab) so a perfect hash table is only 32kB. Also the rules state that no language is ever removed from ISO639-3. I vote for keeping a copy of the database (only the 3-letter codes) in our source tree and testing for validity in Python before inserting into the database.

Reply:
We should only include a subset. We don't want dead/ancient languages, etc.

But sure, I'll rip out the db part...

Reply:
Ah, I see you raised the subset question in https://github.com/Couchers-org/couchers/issues/409 but I forgot to answer there. It is easier to start restrictive and expand the subset than vice versa, so ok :)